db/migrate/20150930183738_migrate_content_hosts.rb

```diff
@@ -1,14 +1,19 @@
 class MigrateContentHosts < ActiveRecord::Migration
-  class Katello::HostCollectionAssociation < ActiveRecord::Base
+  class Katello::SystemRepository < ActiveRecord::Base
+    self.table_name = "katello_system_repositories"
+    belongs_to :system, :inverse_of => :system_repositories, :class_name => 'Katello::System'
+    belongs_to :repository, :inverse_of => :system_repositories, :class_name => 'Katello::Repository'
+  end
+
+  class Katello::SystemHostCollections < ActiveRecord::Base
     self.table_name = "katello_system_host_collections"
   end
 
-  class Katello::ErrataAssociation < ActiveRecord::Base
+  class Katello::SystemErratum < ActiveRecord::Base
     self.table_name = "katello_system_errata"
-  end
 
-  class Katello::RepositoryAssociation < ActiveRecord::Base
-    self.table_name = "katello_system_repositories"
+    belongs_to :system, :inverse_of => :system_errata, :class_name => 'Katello::System'
+    belongs_to :erratum, :inverse_of => :system_errata, :class_name => 'Katello::Erratum'
   end
 
   class Katello::System < ActiveRecord::Base
```

Comment:
It looks like `Hypervisor`s are the only thing extending systems. Are we using `Hypervisor`?